test/tasks/common_test.go

```diff
@@ -87,3 +92,70 @@ func runTaskTestCases(t *testing.T, taskName string, testCases map[string]taskte
 		})
 	}
 }
+
+func checkSonarQualityGate(t *testing.T, ctxt *pipelinectxt.ODSContext, qualityGateFlag bool, wantQualityGateStatus string) {
+
+	filePath := "../../deploy/cd-namespace/chart/values.generated.yaml"
+	f, err := os.Open(filePath)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	APIToken := ""
+	r := bufio.NewReader(f)
+	s, e := Readln(r)
+	for e == nil {
+		if strings.HasPrefix(s, "sonarPassword:") {
+
+			index := strings.Index(s, "'")
+			if index == -1 {
+				t.Fatalf("'sonarPassword' entry is missing in file %s", filePath)
+			}
+
+			APIToken = s[index+1 : len(s)-1]
+			break
+		}
+
+		s, e = Readln(r)
+	}
```

Comment:
I think I'd prefer reading this password from the secret in K8s. Otherwise we depend on a file here which may or may not be there. The secret will always need to be there as it is used by the pipeline. Reading secrets is already implemented and used e.g. in git.go to get the PW for Bitbucket.